tools/tools.go

```diff
@@ -14,25 +14,35 @@ import (
 	"github.com/ollama/ollama/template"
 )
 
+// Sentinel errors for parsing states
+var (
+	ErrPartialPrefix = errors.New("partial prefix detected")
+
+	ErrPrefixNotFound = errors.New("prefix not found")
+
+	ErrInvalidToolCall = errors.New("invalid tool call format")
+
+	ErrAccumulateMore = errors.New("need to accumulate more content")
```

Comment:
@mxyng I ended up adding a few more to better capture the state. LMK if it should just be the accumulate more and then one other error. Just surfacing them on the debug

Reply:
Do we still need this? Should it be exposed?